test/integration/azureSpecialProperties.spec.ts

```diff
@@ -32,36 +32,38 @@ describe("auth validation", () => {
 
     const client = new AzureSpecialPropertiesClient(
       mockCredential,
-      "1234-5678-9012-3456"
+      "1234-5678-9012-3456",
+      { allowInsecureConnection: true }
     );
 
-    const result = await client.apiVersionDefault.getMethodGlobalValid();
-
-    // Validate operation
-    assert.equal(result._response.status, 200);
+    await client.apiVersionDefault.getMethodGlobalValid(responseStatusChecker);
 
     // Validate auth header
-    assert.equal(
-      result._response.request.headers.get("authorization"),
-      "Bearer test-token"
-    );
+    // assert.equal(
```

Comment:
Is there a reason why we cannot check this with onResponse?

Reply:
This will be dealt after the merge as part of this issue: https://github.com/Azure/autorest.typescript/issues/947